config/Cargo.toml

```diff
@@ -31,11 +31,16 @@ displaydoc = { workspace = true }
 derive_more = { workspace = true }
 cfg-if = { workspace = true }
 once_cell = { workspace = true }
+nonzero_ext = "0.3.0"
```

Comment:
we would likely want to use this in the entire project. There are a few places where we could make use of this. I would move this to root `Cargo.toml`